### PR TITLE
Make caller responsible for providing the tile server

### LINKED ISSUE #84: Allow caller to specify customer tile server url
A caller should be able to choose a tile provider of their choice. The tile server's url may be specified via `Map` prop or via theme. In addition, since most tile providers require attribution, the caller should also be able to specify the proper attribution as well.

- Url should be a required string in the format of `'https://{s}.somedomain.com/blabla/{z}/{x}/{y}{r}.png'` [reference](https://leafletjs.com/reference.html#tilelayer)
- Attribution should be a string

Possible API:
```
<Map
  tiles={{
    url: string,
    attribution?: string,
  }}
/>
```

Extend base theme:
```
const base = {
  cluster: { ... },
  pin: { ... },
  popup: { ... },
  tiles: {
    attribution?: string,
    url: string
  }
};
```

### PR DESCRIPTION
**Pull request by halocline**
Closes #84

Caller should have ability to choose and specify a tile server of their choice whether through a service or self hosted.

- Adds `tileServer` prop
- Updates demo-app examples to provide `tileServer`
- Updates docs
- Reverts peerDependencies change. `build.js` already excludes `grommet-leaflet` dependencies from the bundle. Need to test this will still work for projects using React 17.

### CODE REVIEW

**Review by taysea — COMMENTED** [review]
A general question -- should we be uploading the `dist` folder to `main` or should that be in the `.gitignore`?

**Comment by taysea** [comment]
Looks like we lost the CSS overrides for Popup at some point. There is extra padding and the anchor has the leaflet treatment.
 
Screen Shot 2023-11-06 at 9 00 31 AM

File: README.md [hunk 1]
```diff
@@ -59,9 +58,11 @@ Map supports all [Leaflet Map](https://leafletjs.com/reference.html#map) and [Re
 
 In addition, it also supports the following props:
 
-| Prop    | Type     | Notes                                   |
-| ------- | -------- | --------------------------------------- |
-| `theme` | `object` | See [theme structure](#theme-structure) |
+| Prop                                                                          | Type     | Default value | Notes                                                                                                                 |
+| ----------------------------------------------------------------------------- | -------- | ------------- | --------------------------------------------------------------------------------------------------------------------- |
+| `tileLayer`                                                                   | `object` | `undefined`   | { url: `string` in the form of `'https://{s}.somedomain.com/blabla/{z}/{x}/{y}{r}.png'`, attribution: `string` e.g. ` |
```
[the remark this answers is not in the document]
Comment (halocline) [reply README.md:63]: Done.

**Comment by halocline** [comment]
> A general question -- should we be uploading the `dist` folder to `main` or should that be in the `.gitignore`?

Let's try it. I think we needed it at one point, but the set up has since been modified. Added.

**Comment by halocline** [comment]
> Looks like we lost the CSS overrides for Popup at some point. There is extra padding and the anchor has the leaflet treatment.
> 
> Screen Shot 2023-11-06 at 9 00 31 AM

I think we are going to need to revisit the approach to how the CSS reset is being applied to caller apps.

**Review by taysea — APPROVED** [review]

**Pull request merged**